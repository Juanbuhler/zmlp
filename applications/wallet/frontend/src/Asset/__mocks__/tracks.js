const tracks = [
  { label: 'English', kind: 'captions', src: '/webvtt/english.vtt' },
  { label: 'French', kind: 'captions', src: '/webvtt/french.vtt' },
  {
    label: 'gcp-label-detection',
    kind: 'metadata',
    src: '/webvtt/gcp-label-detection.vtt',
  },
  {
    label: 'gcp-object-detection',
    kind: 'metadata',
    src: '/webvtt/gcp-object-detection.vtt',
  },
]

export default tracks
