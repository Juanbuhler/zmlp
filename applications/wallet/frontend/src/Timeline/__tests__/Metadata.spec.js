import TestRenderer, { act } from 'react-test-renderer'

import tracks from '../../Asset/__mocks__/tracks'

import TimelineMetadata from '../Metadata'

const noop = () => {}

const ASSET_ID = 'vZgbkqPftuRJ_-Of7mHWDNnJjUpFQs0C'

describe('<TimelineMetadata />', () => {
  it('should toggle Metadata', () => {
    const component = TestRenderer.create(
      <TimelineMetadata
        videoRef={{
          current: {
            textTracks: {
              0: tracks[2],
              length: 2,
              addEventListener: noop,
              removeEventListener: noop,
            },
          },
        }}
        assetId={ASSET_ID}
      />,
    )

    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Open Metadata' })
        .props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Close Metadata' })
        .props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()
  })
})
