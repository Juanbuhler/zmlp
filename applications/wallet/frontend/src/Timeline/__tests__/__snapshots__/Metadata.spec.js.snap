// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineMetadata /> should toggle Metadata 1`] = `
.emotion-0 {
  margin-right: 4px;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  padding: 4px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #73b61c;
  background-color: transparent;
  font-family: Roboto Condensed;
  text-transform: uppercase;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #2e2e2e;
  color: #73b61c;
}

.emotion-1:hover svg path,
.emotion-1.focus-visible:focus svg path {
  fill: #73b61c;
}

<button
  aria-label="Close Metadata"
  className="emotion-1"
  onClick={[Function]}
  type="button"
>
  <svg
    className="emotion-0"
    color="#73b61c"
    data-file-name="timebasedMetadata.svg"
    height={20}
  />
  Metadata
</button>
`;

exports[`<TimelineMetadata /> should toggle Metadata 2`] = `
.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  padding: 4px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #808080;
  background-color: transparent;
  font-family: Roboto Condensed;
  text-transform: uppercase;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #2e2e2e;
  color: #ffffff;
}

.emotion-1:hover svg path,
.emotion-1.focus-visible:focus svg path {
  fill: #ffffff;
}

.emotion-0 {
  margin-right: 4px;
}

<button
  aria-label="Open Metadata"
  className="emotion-1"
  onClick={[Function]}
  type="button"
>
  <svg
    className="emotion-0"
    color="#808080"
    data-file-name="timebasedMetadata.svg"
    height={20}
  />
  Metadata
</button>
`;
